addons/pinniped/config-controller/controllers/controllers.go

```diff
@@ -76,156 +62,150 @@ func (c *PinnipedController) SetupWithManager(manager ctrl.Manager) error {
-	Concierge          concierge `yaml:"concierge"`
-}
+	cluster, err := utils.GetClusterFromSecret(ctx, c.client, secret)
+	if err != nil {
+		if k8serror.IsNotFound(err) {
+			// when cluster is deleted, secret will get deleted since it has an owner ref
+			log.V(1).Info("cluster is getting deleted, skipping secret reconcile")
+			return nil
+		}
+		log.Error(err, "error getting cluster")
+	}
 
-type concierge struct {
-	Audience string `yaml:"audience,omitempty"`
-}
+	log = log.WithValues(constants.ClusterNamespaceLogKey, cluster.Namespace, constants.ClusterNameLogKey, cluster.Name)
 
-// nolint:funlen,nolintlint // Eh, we can live with a function of this length
-func (c *PinnipedController) reconcileAddonSecret(ctx context.Context, cluster *clusterapiv1beta1.Cluster, pinnipedInfoCM *corev1.ConfigMap) error {
-	log := c.Log.WithValues(constants.NamespaceLogKey, cluster.Namespace, constants.NameLogKey, cluster.Name)
-	// check if cluster is scheduled for deletion, if so, delete addon secret on mgmt cluster
+	// check if cluster is scheduled for deletion, if so, skip reconciling secret
 	if !cluster.GetDeletionTimestamp().IsZero() {
-		log.V(1).Info("cluster is getting deleted, deleting addon secret")
-		if err := c.reconcileClusterDelete(ctx, client.ObjectKeyFromObject(cluster)); err != nil {
-			return fmt.Errorf("failed to reconcile cluster delete: %w", err)
-		}
+		log.V(1).Info("cluster is getting deleted, skipping secret reconcile")
 		return nil
 	}
-	var (
-		supervisorAddress  string
-		supervisorCABundle string
-	)
-
-	identityManagementType := "none"
-	if pinnipedInfoCM.Data != nil {
-		var labelExists bool
-		identityManagementType = "oidc"                                // nolint:goconst
-		supervisorAddress, labelExists = pinnipedInfoCM.Data["issuer"] // TODO: get rid of raw strings...
-		if !labelExists {
-			err := errors.New("could not find issuer")
-			log.Error(err, "error retrieving issuer from pinniped-info configmap")
-			return err
-		}
-		supervisorCABundle, labelExists = pinnipedInfoCM.Data["issuer_ca_bundle_data"] // TODO: get rid of raw strings...
-		if !labelExists {
-			err := errors.New("could not find ca bundle")
-			log.Error(err, "error retrieving ca bundle from pinniped-info configmap")
-			return err
-		}
-	} else {
-		supervisorAddress = ""
-		supervisorCABundle = ""
+
+	if err := c.reconcileDataValues(ctx, secret, cluster, pinnipedInfoCM); err != nil {
+		return err
 	}
 
-	log.V(1).Info("retrieved data from pinniped-info configmap", "supervisorAddress", supervisorAddress, "supervisorCABundle", supervisorCABundle)
+	return nil
+}
 
-	pinnipedAddonSecret := &corev1.Secret{
+func (c *PinnipedController) reconcileDataValues(ctx context.Context, secret *corev1.Secret, cluster *clusterapiv1beta1.Cluster, pinnipedInfoCM *corev1.ConfigMap) error {
+	log := c.Log.WithValues(constants.SecretNamespaceLogKey, secret.Namespace, constants.SecretNameLogKey, secret.Name)
```

Comment:
^I think it actually adds to it, but let me double check.

Reply:
oh geez, sorry, I just reread your comment.  Yeah, I debated on this one... (using the `c.Log` and setting those values again vs. passing in the `log` var as an arg to the function (which I did in some places when the func wasn't associated w/the `PinnipedController`).  I didn't feel like we needed the cluster namespace and name at this point since we'd gotten past getting the cluster and errors that might fall out of that, so I only set the secret name/namespace.  I'm open to passing in the log as an argument since it preserves the values we've set though.  Let me know what you think.

Reply:
adding it is no problem, you think it's better to pass the logger in as an arg to these functions?